### PR TITLE
feat(lang): add triple-quoted block string literals

### PR DESCRIPTION
**Pull request by t0k0sh1**
## Summary
- Add `"""..."""` block string literals for embedding multiline text (Markdown documentation, multi-paragraph messages) into Ry source without per-line prefixes.
- The lexer normalizes the literal per the issue spec — drops the incidental leading/trailing newlines, strips the baseline indent matching the closing delimiter, preserves intentional blank lines, decodes the same escape sequences as regular strings.
- The formatter preserves the triple-quoted form on round-trip via a new `is_block` flag on `StringExpr`; regular strings stay single-quoted. The tree-sitter grammar models the literal via an internal `token(seq(...))` rule (no external scanner required).

## Test plan
- [x] `./build-rust/ry_tests` — 2752 C++ tests pass (new `LexerTest.BlockStringTokens`, `ParserTest.LetBlockStringLiteral` / `RegularStringStaysRegular` / `LetBlockStringMultiline`, `Formatter.BlockStringLiteralFormatting`).
- [x] `./build-rust/ry test -p` — 207 spec files pass including new `tests/spec/block_strings.test.ry` (13 cases covering issue example, blank-line preservation, escape decoding, indented-context regression).
- [x] ASan / TSan via `run-asan.sh` / `run-tsan.sh` — all green.
- [x] `editor/tree-sitter/check.sh` — `pass=159 skip=48 fail=0`; corpus tests `75/75`.
- [x] `fuzz_parser` — 1.7M runs, no crashes.

Closes #1843

## Summary by CodeRabbit

* **New Features**
  * Added triple-quoted block string literals (`...`) for multiline text with indentation normalization, incidental newline trimming, intentional blank-line preservation, and standard escape handling. Block strings round-trip correctly and preserve the expected runtime string value.
* **Editor Support**
  * Updated syntax highlighting and editor parsing to recognize block strings.
* **Formatter**
  * Added block-string-aware formatting to keep multiline indentation stable across reformatting.
* **Tests**
  * Added lexer/parser/formatter coverage for block-string syntax, escaping, and edge cases.
* **Documentation**
  * Documented the new grammar and triple-quoted string behavior.

### CODE REVIEW

File: src/formatter.cpp [hunk 1]
```diff
@@ -156,6 +156,62 @@ std::string Formatter::escapeString(const std::string &s) {
...
+            case '\\': result += "\\\\"; break;
+            case '\0': result += "\\0"; break;
+            case '\r': result += "\\r"; break;
+            case '"':
+                if (i + 2 < s.size() && s[i + 1] == '"' && s[i + 2] == '"') {
+                    result += "\\\"";
+                } else {
+                    result += '"';
+                }
+                break;
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply src/formatter.cpp:181]: Applied the suggested fix (`i + 3 > s.size()` clause) in `Formatter::escapeBlockString` so trailing `"`/`""` runs are escaped and the literal round-trips. Added a `Formatter.BlockStringLiteralFormatting` regression case for value-ending-in-`"` and value-ending-in-`""` (both `verifyFormatting`-checked). The existing embedded-`"""` test now also exercises rule (b) because the third quote of the inner run sits in the last two positions; both escape orderings round-trip identically.

File: src/lexer/lexer.cpp [hunk 2]
```diff
@@ -557,6 +557,102 @@ Token Lexer::readToken() {
...
+            }
+            normalized.append(raw, ls, lineEnd - ls);
+            if (lineEnd >= endIdx) break;
+            i = lineEnd + 1;
+        }
+
+        pos_ += 3; col_ += 3;
+        at_line_start_ = false;
+        return {TokenKind::BlockString, std::move(normalized), line_, startCol};
+    }
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply src/lexer/lexer.cpp:656]: Applied. Added `TokenKind::BlockString` to both value-token guards in `Lexer::readToken` — the `/` regex-vs-division branch (around line 357) and the `.<digit>` float-vs-tuple-access branch (around line 484). New `LexerTest.BlockStringTokens` cases (`PrevKindRegressionSlash`, `PrevKindRegressionDotDigit`) lock in that `x = """a""" / 2` lexes as `Ident Equals BlockString Slash Number` and `x = """a""".0` lexes as `Ident Equals BlockString Dot Number` (not RegexLiteral, not Float).
